src/upload.rs

```diff
@@ -316,9 +320,10 @@ fn complete_registry(opt: &PublishOpt) -> Result<Registry> {
     let pypirc = load_pypirc();
     let (registry_name, registry_url) = if let Some(repository_url) = opt.repository_url.as_deref()
     {
-        let name = match repository_url {
-            PublishOpt::DEFAULT_REPOSITORY_URL => Some("pypi"),
-            PublishOpt::TEST_REPOSITORY_URL => Some("testpypi"),
+        let normalized_url = PublishOpt::normalize_url(repository_url);
```

Comment:
Now that there is only one call to `normalize_url` might as well just call `trim_end_matches` on `registry_url` directly.